pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -76,6 +85,50 @@ func (d ChangeSetWithVersionedRs) Calculate() ([]Change, error) {
 	return allChanges, nil
 }
 
+func (d ChangeSetWithVersionedRs) checkForMaxDurationAnn(existingRsGrouped map[string][]ctlres.Resource,
+	existingNonVerRsGrouped map[string]ctlres.Resource) error {
+
+	var err error
+	for _, res := range d.newRs {
+		if _, found := res.Annotations()[MaxDurationAnnKey]; found {
+			resKey := VersionedResource{res, nil}.UniqVersionedKey().String()
+			if exRes, found := existingNonVerRsGrouped[resKey]; found {
+				err = updateLastRenewedTime(res, exRes)
+			} else if exRes, found := existingRsGrouped[resKey]; found {
+				err = updateLastRenewedTime(res, exRes[len(exRes)-1])
+			} else {
+				return nil
+			}
+		}
+	}
+	return err
```

Comment:
Seems like we are going to override the error while iterating. Imagine a scenario: total 3 iteration of for loop. Error came in second only. In that case, we will send nil bcoz third iteration, err is nil. I am not clear What needs to be done for 2nd iteration resource? should we fail or let the program proceed?

Reply:
As soon as we encountered error for any resource we will throw an error and exit the program. Will correct the logic as well.